ChatWithTree/ChatWithTreeBot.py

```diff
@@ -0,0 +1,664 @@
+#
+# Gramps - a GTK+/GNOME based genealogy program
+#
+# Copyright (C) 2025 Melle Koning
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+import logging
+LOG = logging.getLogger(".")
+
+try:
+    from typing import Dict, Any, List, Optional, Tuple, Pattern, Iterator
+    import os
+    import json
+    import sys
+    import time
+    import re
+    import inspect
+    import litellm
+except ImportError as e:
+    LOG.warning(e)
+    raise Exception("ChatWithTree requires litellm")
```

Comment:
Putting a try/except around such large blocks is not a good practice, and can hide issues.

Reply:
Everything but "litellm" here is included with python.